### PR TITLE
Feature/86dregytx/can config

### PR DESCRIPTION
**Pull request by XanderHayhoe**
merge drivers and config to develop pls

### CODE REVIEW

File: Common/Inc/config.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+const DataSegment MUX1_TEMP = {BMS_TEMP, 1, 1};
+const DataSegment MUX2_TEMP = {BMS_TEMP, 2, 2};
+const DataSegment MUX3_TEMP = {BMS_TEMP, 3, 3};
+const DataSegment MUX4_TEMP = {BMS_TEMP, 4, 4};
+const DataSegment MUX5_TEMP = {BMS_TEMP, 5, 5};
+const DataSegment MUX6_TEMP = {BMS_TEMP, 6, 6};
+// END DEFS FOR THERMISTOR MESSAGE
+
+// BEGIN DEFS FOR ERROR STATES
+const DataSegment MUX1_TEMP_WARN = {BMS_WARN_OVERTEMP, 1, 1};
```
Comment (mahir-mahota) [inline Common/Inc/config.h]: Don't need separate warn segments since the WARN_OFFSET implementation deals with that

File: Common/Inc/config.h [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#define BUFF_SIZE 32
+
+#define WARN_OFFSET 0
+
+//MOTOR CONTROLLER
+#define MOTOR_CONTROLLER_1     0x10F8109A
+#define MOTOR_CONTROLLER_2     0x10F8108D
+//BMS
+#define BMS_TEMP             0x10
+#define BMS_WARN_OVERTEMP    0x06
```
Comment (mahir-mahota) [inline Common/Inc/config.h]: Keep naming consistent with only adding _WARN at the end of the non-warn name. Also, make the ID dependent on the WARN_OFFSET. See the sensor board example

**Review by mahir-mahota — CHANGES_REQUESTED** [review]

File: Common/Inc/config.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+//END MOTOR CONTROLLER BOARD DEFS
+
+// BEGIN BMS DEFS
+// BEGIN DEFS FOR THERMISTOR MESSAGE
+const DataSegment MUX1_TEMP = {BMS_TEMP, 1, 1};
+const DataSegment MUX2_TEMP = {BMS_TEMP, 2, 2};
+const DataSegment MUX3_TEMP = {BMS_TEMP, 3, 3};
+const DataSegment MUX4_TEMP = {BMS_TEMP, 4, 4};
+const DataSegment MUX5_TEMP = {BMS_TEMP, 5, 5};
+const DataSegment MUX6_TEMP = {BMS_TEMP, 6, 6};
```
Comment (mahir-mahota) [inline Common/Inc/config.h]: Need an error code byte

File: Common/Inc/config.h (hunk 1 above)
Comment (XanderHayhoe) [reply Common/Inc/config.h]: I think we still do. Don't we need the other boards to recognize the ID?

**Pull request merged**